cmake/onnxruntime_providers.cmake

```diff
@@ -950,6 +953,57 @@ if (onnxruntime_USE_COREML)
   endif()
 endif()
 
+if (onnxruntime_USE_WEBNN)
+
+  if (onnxruntime_MINIMAL_BUILD AND NOT onnxruntime_EXTENDED_MINIMAL_BUILD)
+    message(FATAL_ERROR "WebNN EP can not be used in a basic minimal build. Please build with '--minimal_build extended'")
+  endif()
+
+  if (WIN32)
+    set(DISABLED_WARNINGS_FOR_WEBNN /wd4305 /wd4244)
```

Comment:
Please help check which flags are legacy for native WebNN EP build, and remove them.